### PR TITLE
add attachment support to Mailgun action

### PR DESCRIPTION
**Pull request by igor-makarov**
### Checklist
- [✅ ] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [✅] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [✅] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [✅] I've updated the documentation if necessary.

### Description
Allow using `:attachment` param to add files to email.

### Motivation and Context
As a developer, I wanted to receive an attachment that contains my UI test results.

### CODE REVIEW

File: fastlane/lib/fastlane/actions/mailgun.rb [hunk 1]
```diff
@@ -86,7 +86,12 @@ def self.available_options
                                       env_name: "MAILGUN_REPLY_TO",
                                       description: "Mail Reply to",
                                       optional: true,
-                                      is_string: true)
+                                      is_string: true),
+          FastlaneCore::ConfigItem.new(key: :attachment,
+                                      env_name: "MAILGUN_ATTACHMENT",
+                                      description: "Mail Attachments",
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/mailgun.rb]: Could you update the description to describe better on what to pass? An array? One item as string? The data or the file path(s)?

File: fastlane/lib/fastlane/actions/mailgun.rb [hunk 2]
```diff
@@ -114,6 +119,13 @@ def self.mailgunit(options)
         unless options[:reply_to].nil?
           params.store(:"h:Reply-To", options[:reply_to])
         end
+
+        unless options[:attachment].nil?
+          attachment_filenames = [*options[:attachment]]
+          attachments = attachment_filenames.map { |filename| File.new(filename, 'rb') }
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/mailgun.rb]: Why the rb?

File: fastlane/lib/fastlane/actions/mailgun.rb [hunk 3]
```diff
@@ -158,7 +170,8 @@ def self.example_code
             message: "Mail Body",
             app_link: "http://www.myapplink.com",
             ci_build_link: "http://www.mycibuildlink.com",
-            template_path: "HTML_TEMPLATE_PATH"
+            template_path: "HTML_TEMPLATE_PATH",
+            attachment: "build/changelog.txt"
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/mailgun.rb]: Why build/changelog?

**Review by KrauseFx — CHANGES_REQUESTED** [review]
Thanks for the contribution. File attachment seems like a great feature for fastlane :+1:

File: fastlane/lib/fastlane/actions/mailgun.rb (hunk 2 above)
Comment (igor-makarov) [inline fastlane/lib/fastlane/actions/mailgun.rb]: I'm not very familiar with the distinction, but this is how it appears in [rest-client's docs](https://github.com/rest-client/rest-client#multipart) so I didn't think too much of it. 

From documentation it seems like the `b` option causes no CRLF conversion to be performed, which is obviously the case for attachments.

File: fastlane/lib/fastlane/actions/mailgun.rb (hunk 3 above)
Comment (igor-makarov) [inline fastlane/lib/fastlane/actions/mailgun.rb]: You're right, this is less clear. Will revise.

File: fastlane/lib/fastlane/actions/mailgun.rb (hunk 1 above)
Comment (igor-makarov) [inline fastlane/lib/fastlane/actions/mailgun.rb]: It's filename or array of filenames. I'll update.

**Comment by igor-makarov** [comment]
I updated the requested stuff.

File: fastlane/lib/fastlane/actions/mailgun.rb (hunk 2 above)
Comment (hjanuschka) [inline fastlane/lib/fastlane/actions/mailgun.rb]: its "read-binary", i think they do it this way in the documentation as if you don't use "rb" you get in troubles with non character files.

Comment (igor-makarov) [inline fastlane/lib/fastlane/actions/mailgun.rb]: So do you agree that it's right in this case?

**Review by hjanuschka — APPROVED** [review]

**Review by KrauseFx — APPROVED** [review]

**Comment by KrauseFx** [comment]
Awesome, thanks you both! 👍

**Pull request merged**